File: C/Grace.c
#include <stdio.h>

#define FORMAT "#include <stdio.h>%1$c%1$c#define FORMAT %2$c%3$s%2$c%1$c#define SAY(f, s) fprintf(f, s, 10, 34, s);%1$c#define MAIN() int main(void){ FILE *f = fopen (%2$cGrace_kid.c%2$c, %2$cw%2$c); if (f == NULL) return 1; SAY(f, FORMAT) fclose(f); return 0; }%1$c%1$c/*%1$c  this program writes its source code in a file%1$c*/%1$c%1$cMAIN();%1$c"
#define SAY(f, s) fprintf(f, s, 10, 34, s);
#define MAIN() int main(void){ FILE *f = fopen ("Grace_kid.c", "w"); if (f == NULL) return 1; SAY(f, FORMAT) fclose(f); return 0; }

/*
  this program writes its source code in a file
*/

MAIN();

File: Python/Grace.py
FORMAT = 'FORMAT = {quote}{format}{quote}{lf}{lf}CODE = """{lf}try:{lf}{ht}with open("Grace_kid.py", "w+") as f:{lf}{ht}{ht}print(FORMAT.format(lf=chr(10), ht=chr(9), quote=chr(39), format=FORMAT), file=f){lf}except OSError as error:{lf}{ht}print(error){lf}"""{lf}{lf}MAIN = exec{lf}{lf}"""{lf}{ht}this program writes its source code in a file{lf}"""{lf}{lf}MAIN(CODE)'

CODE = """
try:
	with open("Grace_kid.py", "w+") as f:
		print(FORMAT.format(lf=chr(10), ht=chr(9), quote=chr(39), format=FORMAT), file=f)
except OSError as error:
	print(error)
"""

MAIN = exec

"""
	this program writes its source code in a file
"""

MAIN(CODE)
